vowpalwabbit/v_array.h

```diff
@@ -76,6 +76,21 @@ struct v_array<T, typename std::enable_if<std::is_trivially_copyable<T>::value>:
     memmove(&_begin[idx + width], &_begin[idx], (size() - (idx + width)) * sizeof(T));
   }
 
+  void resize_no_initialize(size_t old_size, size_t length)
+  {
+    // if new length is smaller than current size destroy the excess elements
+    for (auto idx = length; idx < old_size; ++idx) { _begin[idx].~T(); }
+    reserve(length);
+    _end = _begin + length;
+  }
+
+  void copy_into_this(const v_array<T>& src)
```

Comment:
should this just be called `copy`?

Reply:
I went with `copy_into_this` as there is no `dest` argument so it makes it clear where it is ending up